app/controllers/application_controller.rb

```diff
@@ -2,4 +2,14 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  helper_method :current_user
+
+  def current_user
+    #look up once
+    @current_user ||= User.find(session[:uid]) if session[:uid]
```

Comment:
fixed and committed 

----- Original Message -----

From: "Levi Cole" <notifications@github.com> 
To: "TeriPastorino/KnoWater" <KnoWater@noreply.github.com> 
Cc: "TeriPastorino" <tpasto@comcast.net> 
Sent: Tuesday, May 12, 2015 7:56:06 AM 
Subject: Re: [KnoWater] Signup (#7) 



In app/controllers/application_controller.rb : 
> @@ -2,4 +2,14 @@ class ApplicationController < ActionController::Base
>    # Prevent CSRF attacks by raising an exception.
>    # For APIs, you may want to use :null_session instead.
>    protect_from_forgery with: :exception
> +  helper_method :current_user
> +
> +  def current_user
> +    #look up once
> +    @current_user ||= User.find(session[:uid]) if session[:uid] 


This method was actually ok before. The only thing that needed to change was line 4 of SessionsController. The key in the session here, should match the key in the session on line 4 in the SessionsController. 

— 
Reply to this email directly or view it on GitHub .